code/obj/item/rcd.dm

```diff
@@ -293,6 +314,66 @@ Broken RCD + Effects
 						new map_settings.windows(get_turf(A))
 						log_construction(user, "builds a window")
 						return
+			if (RCD_MODE_LIGHTBULBS)
+				if (istype(A, /turf/simulated/wall))
+					if((locate(/obj/machinery/light) in A) || (locate(/obj/machinery/light) in get_turf(user)))
+						boutput(user, "There's already a light there!") // stacking lights simply can't be good for the environment
+						return
+					var/dir
+					for (var/d in cardinal)
+						if (get_step(user,d) == A)
+							dir = d
+							break
+					if(!dir) // lights only apply themselves if standing at a cardinal direction from the wall
+						boutput(user, "You can't seem to reach that part of \the [A]. Try standing right up against it.")
+						return
+					var/turf/simulated/wall/W = A
+					if (do_thing(user, W, "attaching a light bulb fixture to \the [W]", matter_create_light_fixture, 2 SECONDS))
+						var/datum/material/M
+						if(W.material)
+							M = W.material
```

Comment:
I'm not sure if I like this and other similar lines as it could be used to duplicate materials. I admit that it's at the steep price of RCD ammo but I think I'd rather be safe and use `material_name` exclusively.

Reply:
RCD ammo is surprisingly cheap, molitz apparently works for manufacturing it. And I just tested it, and yeah that's definitely an issue. I'll set everything to use the RCD's `material_name`.